packages/ui/react-ui-stack/src/components/Deck.stories.tsx

```diff
@@ -0,0 +1,115 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import '@dxosTheme';
+
+import { BookBookmark, StackSimple } from '@phosphor-icons/react';
+import React, { useState } from 'react';
+
+import { faker } from '@dxos/random';
+import { AttentionProvider, PlankHeading, plankHeadingIconProps, Deck } from '@dxos/react-ui-deck';
+import { Mosaic } from '@dxos/react-ui-mosaic';
+import { withTheme } from '@dxos/storybook-utils';
+
+import { type StackSectionItem } from './Section';
+import { Stack } from './Stack';
+
+faker.seed(3);
+
+const Toolbar = () => {
+  return (
+    <div role='toolbar' className='flex gap-2 pis-[--rail-size]'>
+      <p>File</p>
+      <p>Edit</p>
+      <p>View</p>
+    </div>
+  );
+};
+
+const SimpleContent = ({ data }: { data: { title?: string; body?: string } }) => (
+  <>
+    <div className='text-xl mlb-2'>{data.title}</div>
+    <p>{data.body}</p>
+  </>
+);
+
+const rollItems = (n: number): StackSectionItem[] => {
+  return [...Array(n)].map(() => ({
+    id: faker.string.uuid(),
+    icon: BookBookmark,
+    object: {
+      id: faker.string.uuid(),
+      title: faker.lorem.words(3),
+      body: faker.lorem.paragraph(),
+    },
+  }));
+};
+
+const DemoStackPlank = ({ toolbar }: { toolbar?: boolean }) => {
+  const [items] = useState(rollItems(12));
+  return (
+    <>
+      <PlankHeading.Root classNames='pli-px'>
+        <PlankHeading.Button>
+          <StackSimple {...plankHeadingIconProps} />
+        </PlankHeading.Button>
+        <PlankHeading.Label>{faker.lorem.words(3)}</PlankHeading.Label>
+      </PlankHeading.Root>
+      {toolbar && <Toolbar />}
+      <Stack
+        items={items}
+        SectionContent={SimpleContent}
```

Comment:
how is the collapsed section component defined?
e.g., can you demo a collapsed section that just shows the first n line(s) of the editor (so that it doesn't jump).

Reply:
Stack maintains a controllable state for its sections’ collapsed states, currently. See https://github.com/dxos/dxos/blob/main/packages/ui/react-ui-stack/src/components/Stack.tsx#L73-L77

Collapsed sections only display their entity’s title rather than delegating to any surfaces. I think I’d use a different term for a feature that could be described as “sections render their content within a fixed height”, maybe ‘truncate’? This feature is able to scale to different levels if we like, so title-only collapsing can be supported alongside truncation collapsing.